### PR TITLE
Clue 0.5.0. Actually 0.6.0, probably. Make that 0.7.0.

### PR DESCRIPTION
**Pull request by rpiaggio**
Adapts code to the new version of clue. In particular, this allows for an authorization token to be passed upon establishing an Apollo WS connection.

Build will fail until clue 0.5.0 is released, thus I'm marking this PR as draft until then.

### CODE REVIEW

File: common/src/main/scala/explore/AppMain.scala [hunk 1]
```diff
@@ -106,6 +108,7 @@ trait AppMain extends IOApp {
       appConfig <- fetchConfig
       _         <- logger.info(s"Config: ${appConfig.show}")
       ctx       <- AppContext.from[IO](appConfig)
+      _         <- ctx.clients.odb.connect(JsonObject("Authorization" -> "Bearer ...".asJson))
```
Comment (cquiroz) [inline common/src/main/scala/explore/AppMain.scala:111]: I guess we need to stick the token instead of `...`?

File: targeteditor/src/main/scala/explore/targeteditor/MagnitudeForm.scala [hunk 2]
```diff
@@ -82,7 +81,7 @@ object MagnitudeForm {
               )
             )
 
-          Form(size = Small)(
+          React.Fragment(
```
Comment (cquiroz) [inline targeteditor/src/main/scala/explore/targeteditor/MagnitudeForm.scala:84]: Why this change? Seems unrelated

Comment (rpiaggio) [reply targeteditor/src/main/scala/explore/targeteditor/MagnitudeForm.scala:84]: Yes, it's unrelated. There was a `Form` inside of another `Form` (from `TargetBody`).

File: common/src/main/scala/explore/AppMain.scala (hunk 1 above)
Comment (rpiaggio) [reply common/src/main/scala/explore/AppMain.scala]: Exactly. I just pushed a new commit with further changes to clue which allow us to specify a different token in each reconnection.

File: common/src/main/scala/explore/AppMain.scala [hunk 3]
```diff
@@ -101,11 +105,30 @@ trait AppMain extends IOApp {
...
+
     for {
       _         <- logger.info(s"Git Commit: [${BuildInfo.gitHeadCommit.getOrElse("NONE")}]")
       appConfig <- fetchConfig
       _         <- logger.info(s"Config: ${appConfig.show}")
       ctx       <- AppContext.from[IO](appConfig)
+      _         <- ctx.clients.odb
+                     .connect(
+                       connectParameters,
+                       reconnectionStrategy.asInstanceOf[ReconnectionStrategy[IO, ctx.clients.odb.CE]].some
```
Comment (rpiaggio) [inline common/src/main/scala/explore/AppMain.scala:130]: I'm working of ways to improve the types in `clue` so that this cast is not necessary. I will investigate using `sttp` but it will take a while, so I'm pushing this in the meantime not to hold things up.

File: common/src/main/scala/explore/components/ConnectionsStatus.scala [hunk 4]
```diff
@@ -54,8 +55,12 @@ object ConnectionsStatus {
         <.span(
           ctx.clients.ExploreDBConnectionStatus(renderStatus("Hasura DB")).when(false),
           ctx.clients.ODBConnectionStatus(renderStatus("ODB")),
-          Button(size = Tiny)(^.onClick --> ctx.clients.odb.close().runAsyncCB)("Close ODB")
-            .when(false)
+          Button(size = Tiny)(
+            ^.onClick --> ctx.clients.odb
+              .disconnect(WSJSCloseParams(code = 4000).asInstanceOf[ctx.clients.odb.CP])
```
Comment (rpiaggio) [inline common/src/main/scala/explore/components/ConnectionsStatus.scala:60]: I'm working of ways to improve the types in `clue` so that this cast is not necessary. I will investigate using `sttp` but it will take a while, so I'm pushing this in the meantime not to hold things up.

File: common/src/main/scala/explore/AppMain.scala [hunk 5]
```diff
@@ -55,7 +59,7 @@ trait AppMain extends IOApp {
   def runIOApp(): Unit = main(Array.empty)
 
   override final def run(args: List[String]): IO[ExitCode] = {
-    japgolly.scalajs.react.extra.ReusabilityOverlay.overrideGloballyInDev()
+    // japgolly.scalajs.react.extra.ReusabilityOverlay.overrideGloballyInDev()
```
Comment (cquiroz) [inline common/src/main/scala/explore/AppMain.scala]: Should we restore this?

File: common/src/main/scala/explore/AppMain.scala [hunk 6]
```diff
@@ -101,11 +105,30 @@ trait AppMain extends IOApp {
       IO.fromFuture(httpCall).map(_.body)
     }
 
+    val fetchToken: IO[String] = IO("...")
+
+    val connectParameters: IO[JsonObject]                          =
+      fetchToken.map(token => JsonObject("Authorization" -> s"Bearer $token".asJson))
+
+    val reconnectionStrategy: ReconnectionStrategy[IO, CloseEvent] =
+      ReconnectionStrategy(maxAttempts = 100,
```
Comment (cquiroz) [inline common/src/main/scala/explore/AppMain.scala:114]: Any reason for a 100? Or is it reset once you reconnect?

**Review by cquiroz — APPROVED** [review]
LGTM

**Comment by swalker2m** [comment]
Would either of you take a look at the [corresponding updates in the server for `0.5.0`](https://github.com/gemini-hlsw/lucuma-tmp-api/pull/96) to make sure it is correct and compatible?

File: common/src/main/scala/explore/AppMain.scala (hunk 5 above)
Comment (rpiaggio) [reply common/src/main/scala/explore/AppMain.scala]: Good catch

File: common/src/main/scala/explore/AppMain.scala (hunk 6 above)
Comment (rpiaggio) [reply common/src/main/scala/explore/AppMain.scala:114]: The attempt counter is indeed reset upon a successful reconnection.

I just put a high number there for now. We have to decide what reconnection strategy we will use, we probably want an exponential backoff instead of a linear one.

How does it sound to have a capped exponential backoff with infinite attempts?

Comment (cquiroz) [reply common/src/main/scala/explore/AppMain.scala:114]: Sure that sounds like what the seqexec does
Would you do that in a separate PR?

Comment (rpiaggio) [reply common/src/main/scala/explore/AppMain.scala:114]: Didn't think of copying what we do in Seqexec. I'm sending a new commit with changes for clue 0.6.0 and I'm including the same logic as there.
